bindings/node/.evergreen/prebuild.sh

```diff
@@ -1,35 +1,52 @@
 #!/usr/bin/env bash
 
 set -o errexit
+set +o xtrace
+
+# FLE platform matrix (as of Oct 6th 2021)
+# macos   x86_64 (compiled on 10.14)
+# windows x86_64 (compiled on vs2017)
+# linux   x86_64
+# linux   s390x
+# linux   arm64
 
 get_version_at_git_rev () {
   local REV=$1
-  local VERSION=$(node -r child_process -e "console.log(JSON.parse(child_process.execSync('git show $REV:./package.json', { encoding: 'utf8' })).version);")
-  echo $VERSION
+  local VERSION
+  VERSION=$(node -r child_process -e "console.log(JSON.parse(child_process.execSync('git show $REV:./package.json', { encoding: 'utf8' })).version);")
+  echo "$VERSION"
 }
 
 run_prebuild() {
-  if [[ -z $NODE_GITHUB_TOKEN ]];then
-    echo "No github token set. Cannot run prebuild."
-    exit 1
-  else
-    echo "Github token detected. Running prebuild."
-    npm run prebuild -- -u $NODE_GITHUB_TOKEN
-    echo "Prebuild's successfully submitted"
-  fi
+  if [ -z ${NODE_GITHUB_TOKEN+omitted} ]; then echo "NODE_GITHUB_TOKEN is unset" && exit 1; fi
+  echo "Github token detected. Running prebuild."
+  npm run prebuild -- -u "$NODE_GITHUB_TOKEN"
+  echo "Prebuild's successfully submitted"
 }
 
 VERSION_AT_HEAD=$(get_version_at_git_rev "HEAD")
 VERSION_AT_HEAD_1=$(get_version_at_git_rev "HEAD~1")
 
-if [[ ! -z $NODE_FORCE_PUBLISH ]]; then
-  echo '$NODE_FORCE_PUBLISH detected'
+if [[ -n $NODE_FORCE_PUBLISH ]]; then
+  echo "\$NODE_FORCE_PUBLISH=${NODE_FORCE_PUBLISH} detected"
   echo "Beginning prebuild"
   run_prebuild
-elif [[ $VERSION_AT_HEAD != $VERSION_AT_HEAD_1 ]]; then
+elif [[ "$VERSION_AT_HEAD" != "$VERSION_AT_HEAD_1" ]]; then
   echo "Difference is package version ($VERSION_AT_HEAD_1 -> $VERSION_AT_HEAD)"
   echo "Beginning prebuild"
-  run_prebuild
+
+  if [[ "$OS" == "LINUX" ]]; then
+    # We're on linux should be able to use uname -m
+    ARCH=$(uname -m)
+    if [[ "$ARCH" == "x86_64" && $DISTRO_ID == "rhel70-small" ]]; then
```

Comment:
Fwiw, I mentioned `$DISTRO_ID` because it already implies the specfific architecture – different architectures will have different distro ids, x86_64 just happens to be the implied default

Reply:
Yeah, makes sense – the conditional here is redundant because its second part implies the first part.

Also, I do think you want to have a final `else` that is just `npm run prebuild`, like the `Will prebuild without submit` part below?

Reply:
Yea that makes sense, I think I've cleaned it up to be clearer now 🙂